pkg/querier/block.go

```diff
@@ -46,6 +46,7 @@ func NewBlockQuerier(cfg tsdb.Config, logLevel logging.Level, r prometheus.Regis
 		return nil, err
 	}
 	b.us = us
+	r.MustRegister(us.tsdbMetrics)
```

Comment:
A couple of things:
1. May be a good practice to register metrics only if `r != nil` (recently I started doing it, cause sometimes in tests we don't want to pass a registerer)
2. I would group `r.MustRegister(b.syncTimes)` here